src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -199,19 +200,55 @@ public bool IsConnected
             }
         }
 
+        public static IPEndPoint GetHandshakedEndPoint(INetworkPeer peer)
+        {
+            if (peer.Inbound && peer.State == NetworkPeerState.HandShaked)
+            {
+                IPEndPoint addressFrom = peer.PeerVersion?.AddressFrom.MapToIpv6();
+
+                if (addressFrom != null && !addressFrom.Address.Equals(IPAddress.IPv6Any))
+                {
+                    if (peer.PeerEndPoint.Address.IsRoutable(false) || !addressFrom.Address.IsRoutable(false))
```

Comment:
Also, the previous code in `PeerAddressManagerBehaviour` did not contain routable checks?

Reply:
The routable checks ensure that we don't use non-routable addresses in favor of routable ones. It also helps classify the ip address we are using to compare, so that we use the correct comparison approach - i.e. local to local and remote to remote.